### PR TITLE
Add atom table growth monitoring to Beam skill

### PR DESCRIPTION
**Pull request by bradleygolden**
Implements atom leak detection with historical tracking. Atoms are never garbage collected, so atom exhaustion is fatal and unrecoverable.

New callbacks:
- beam_atom_growth_rate/1: Track growth rate over time
- beam_atom_leak_detected/0: Detect suspected leaks
- beam_check_atom_safety/0: Review dangerous patterns
- beam_node_name_atoms/0: Explain enumeration limitations

Components:
- AtomStore GenServer for periodic atom count sampling (60s intervals)
- Growth rate analysis with urgency classification
- Educational warnings about binary_to_atom, list_to_atom, xmerl

Zero production impact: Read-only atom counting via :erlang.system_info

555 tests passing

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,19 +9,25 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Message queue overload detection: `beam_queue_processes/1`, `beam_queue_growth/2`, `beam_queue_stats/0`
 - Operators and Coordinator are now always-running supervised processes
 - New `Operator.run_async/3` for running analysis in the background with progress notifications
 - Multiple analysis requests to the same operator are queued and processed in order
 - Google AI (Gemini) provider support for integration tests
 
+### Fixed
+
+- Division by zero edge case in `beam_atom_growth_rate/1` when historical samples have identical timestamps
+
```
Comment (bradleygolden) [inline CHANGELOG.md]: Did you fix this bug as a change against origin main? Is this necessary?
